src/Component/Symfony/Routing/Factory/DeleteOperationRoutePathFactory.php

```diff
@@ -28,10 +29,7 @@ public function createRoutePath(HttpOperation $operation, string $rootPath): str
         $identifier = $operation->getResource()?->getIdentifier() ?? 'id';
 
         if ($operation instanceof DeleteOperationInterface) {
-            $path = match ($shortName) {
-                'delete' => '',
-                default => '/' . $shortName,
-            };
+            $path = $operation instanceof ApiOperationInterface && 'delete' === $shortName ? '' : '/' . $shortName;
```

Comment:
I have mixed feelings about it. The fix is needed, but in API this delete suffix should be required only in **non** API context. Once we are in API context, it is okay to have exactly the same URL but with different HTTP method. This logic is required only when we have a form, that will send a POST request with hidden DELETE notation. 

What I would recommend is to reverse this condition so we add `delete` only for non ApiOperationInterface

Reply:
With my proposal, we have these behaviors.

**Non API**
```php
#[Delete]
#[Delete(name: 'remove')]
```

```
app_book_delete             DELETE        /admin/books/delete 
app_book_remove             DELETE        /admin/books/remove 
```

**API**
```php
#[Delete]
#[Delete(name: 'remove')]
```

```
app_book_delete             DELETE        /admin/books 
app_book_remove             DELETE        /admin/books/remove 
```

IMHO, this is ok like this.